# 式の計算（加減乗除）

In [1]:
# 加算
3+7

10

In [2]:
# 減算
8-5

3

In [3]:
# 乗算
6*5

30

In [4]:
# 除算
9/3

3.0

In [5]:
# 余り
143%3

2

In [6]:
# xのn乗 (x ** n で表す)
3**10

59049

# 関数print()を使用した表示
(print()は、最終的に何かを表示させる以外にも、長いプログラムを書いたときに、処理した内容を表示させることでデバッグに使用できる。)

In [7]:
print('Hello, world!')

Hello, world!


In [8]:
x = 'Hello, world!'
print(x)
y = 'It is cold, today'
print(y)

Hello, world!
It is cold, today


In [9]:
print ('3 + 7 = ', 3+7)

3 + 7 =  10


In [10]:
# 変数xに値を格納
x = 6*5
print(x)
y = 5*8
print(y)

30
40


In [11]:
print('2.5^{2}=', 2.5**2)
print('3.14^{3}=', 3.14**3)
print('6.25^{5}=', 6.25**5)


2.5^{2}= 6.25
3.14^{3}= 30.959144000000002
6.25^{5}= 9536.7431640625


#Latexコマンドによる累乗表示 
#(この表示はCodeモード。Markdownモードに変更し、以下の最初の"#"を外すことで、表示可能)

$2.5^{2}= 6.25$

$3.14^{3}= 30.959144000000002$

# リスト: 値を並べたもの（数値、文字列ともに格納可能）

In [12]:
# 数値のリスト(リストのIDは、"0"から始まることに注意)
a = [5, 6, 7, 8, 9, 10]
print(a[0], a[1], a[2], a[3], a[4], a[5])

# 指定した範囲のリストも取り出せる（後半の数字は、「未満」の意味でとらえると良い）
print(a[2:4])

# 負の数も指定可能
# 後ろから2番目
print(a[-2])

# 前から2番目～後ろから3番目までの範囲
print(a[1:-2])

5 6 7 8 9 10
[7, 8]
9
[6, 7, 8]


In [13]:
# リストの入れ子構造
a = [5, 6, [7, 8], 9]
print(a[2])

[7, 8]


In [14]:
# 文字列のリスト
animals = ['dog', 'cat', 'swallow', 'lion', 'elephant']
print(animals)
print(animals[1])
print(animals[3])

['dog', 'cat', 'swallow', 'lion', 'elephant']
cat
lion


In [15]:
# 数値と文字列の混在リスト（例：名前と年齢）
mixed_list = ['Sato', 32, 'Suzuki', 27, 'Takahashi', 53, 'Tanaka', 45]
print(mixed_list[1])
print(mixed_list[4])

32
Takahashi


In [16]:
# 入れ子構造を利用して、上記のリストの「名前」と「年齢」をペアにして格納することもできる
mixed_list = [['Sato', 32], ['Suzuki', 27], ['Takahashi', 53], ['Tanaka', 45]]
print(mixed_list)
print(mixed_list[2])

# 3番目の要素の中の1番目の要素
print(mixed_list[2][0])

# 4番目の要素の中の2番目の要素
print(mixed_list[3][1])

[['Sato', 32], ['Suzuki', 27], ['Takahashi', 53], ['Tanaka', 45]]
['Takahashi', 53]
Takahashi
45


# 関数とメソッド

In [17]:
# 関数: あるオブジェクトの値に依存して、決まる値を返すもの
# メソッド: オブジェクトに対して、何らかの処理を行うもの
# len(): リストの長さを返す関数
# sort(): リストの要素を、小さい順(昇順, ascending order)に並べ替えるメソッド
# sort(reverse=True): リストの要素を、大きい順(降順, descending order)に並べ替えるメソッド 

a = [3, 6, 1, 8, 5]
print('もともとのリスト', a)

# リストaの長さを返す
print('長さ:', len(a))

# リストaを昇順に並べ替える
a.sort()
print('昇順:', a)

# リストaを降順に並べ替える
a = [3, 6, 1, 8, 5]
a.sort(reverse=True) # reverse=True を指定することで、降順に並べ替えられる
print('降順:', a)

もともとのリスト [3, 6, 1, 8, 5]
長さ: 5
昇順: [1, 3, 5, 6, 8]
降順: [8, 6, 5, 3, 1]


In [18]:
# 【補足】
# いったん、sort()を使用して昇順にした後、reverse()を使用して降順に並べ替えることもできる
# reverse() は、リストの要素を逆順に並べ替えるメソッド
print('【補足】')
a = [3, 6, 1, 8, 5]
print('もともとのリスト', a)
a.sort()
print('sort()を使用して昇順:', a)
a.reverse()
print('さらにreverse()を使用して降順:', a)

【補足】
もともとのリスト [3, 6, 1, 8, 5]
sort()を使用して昇順: [1, 3, 5, 6, 8]
さらにreverse()を使用して降順: [8, 6, 5, 3, 1]


In [19]:
# リストanimalsに、'horse'を加える 
# append(): 既存のリストに要素を追加するメソッド
animals = ['dog', 'cat', 'swallow', 'lion', 'elephant']
print(animals)

animals.append('horse')
print(animals)

['dog', 'cat', 'swallow', 'lion', 'elephant']
['dog', 'cat', 'swallow', 'lion', 'elephant', 'horse']


In [20]:
# 乱数の生成 (1)
import random
random.seed() # 基本乱数生成器の初期化（現在のシステム時間を使用。乱数使用時に最初に行なう必要がある）
for n in [1,2,3,4,5]:
    print(random.random()) # random.random(): 0.0～1.0 の範囲のランダムな浮動小数点数を返す

0.9980909020014521
0.5499017179911324
0.19796838948216944
0.2817699571896982
0.5623818988066781


In [21]:
# 乱数の生成 (2)
import random
random.seed() # 基本乱数生成器の初期化（現在のシステム時間を使用）
for n in [1,2,3,4,5]:
    print(random.randint(1,100)) # random.randint(a,b): a<=N<=b を満たすランダムな整数Nを返す

35
99
1
47
77
